runtime/lib/ttnn/operations/layout/to_layout.cpp

```diff
@@ -35,8 +35,8 @@ void run(const ::tt::target::ttnn::ToLayoutOp *op, ProgramContext &context) {
   std::optional<::ttnn::MemoryConfig> memoryConfig = std::nullopt;
   ::ttnn::Device *device = nullptr;
 
-  if (op->dtype() != ::tt::target::DataType::None) {
-    dtype = ::tt::runtime::ttnn::utils::toTTNNDataType(op->dtype());
+  if (op->dtype()) {
```

Comment:
Will this work for `Float32`?

Reply:
Yes it will because the enum is wrapped in a flatbuffers::Optional (because we're assigning null in the fbs file), so this check will solely check if the Optional has a value and is not a flatbuffers::nullopt. That's also why I have to dereference the dtype in the next line.